### DISCUSSION TITLE
How to set the path to look inside the exe

### DISCUSSION THREAD

**Discussion by mega6453** [yao-pkg/pkg#44] [Q&A]
May be my question won't be clear. Because I'm not a expert in this nodejs or pkg. 

What I want to do is : Pack appium(mobile automation library) and it's drivers(android,ios) inside the node exe file and ship it to the user's machine. Appium is the main app and it will use the drivers. So, appium has a code that says where to look for the drivers in the following path : "\node_modules\@appium\support\build\lib\env.js". Following is the code: exports.DEFAULT_APPIUM_HOME = path_1.default.resolve((0, os_1.homedir)(), '.appium');

As per the above code, it looks for the drivers in the user's home directory. But I want it to use the driver which is packed inside the exe file. how to do that?

Below is my package.json which includes all the files in the node_modules(I HOPE):

{
  "name": "appium",
  "version": "2.5.1",
  "description": "Appium Server",
  "main": "index.js",
  "scripts": {
    "test": "appium"
  },
  "bin": {
    "appium": "./index.js"
  },
  "author": "Test user",
  "license": "ISC",
  "dependencies": {
    "appium": "^2.5.1",
    "appium-uiautomator2-driver": "^3.0.2",
    "appium-xcuitest-driver": "^7.3.1"
  },
  "pkg": {
    "targets": [
      "node16-win-x64"
    ],
    "assets": [
      "./node_modules/**/*"
    ],
    "outputPath": "dist"
  }
}

**Comment by mega6453** [comment]
@robertsLando Any help please

**Reply by robertsLando** [reply]
You should not copy the entire node_modules folder on assets but just the things you want to be shipped with the code that are not picked up automatically by pkg. When you run pkg it traverse all your code starting from the provided entrypoint and looks for the require statements and follows them in order to retrieve the required modules to bundle.

> As per the above code, it looks for the drivers in the user's home directory. But I want it to use the driver which is packed inside the exe file. how to do that?

If you have exe files or else that need to be shipped those should be added as assets and then you should read them from snapshot and copy them to the local directory where you want to place them

**Reply by mega6453** [reply]
@robertsLando First, thanks for the reply.

As I'm new to this, for more clarity asking bit more.
What I Understood is : I've to add only the appium-uiautomator2-driver in the assets section..

Question I've:
How to read and copy the files from snapshot to local directory?

Help please.. Thanks.

**Reply by robertsLando** [reply]
You can use fs.readFile to read and fs.writeFile to write to local directory. When using the readFile you should use `__dirname` as base and when writing locally you can use an absolute path or using a relative one with `process.cwd()`. Check https://github.com/yao-pkg/pkg#snapshot-filesystem for more info

**Reply by mega6453** [reply]
Thanks for the info.. I'll try this..
